### PR TITLE
add DAOs

### PR DESCRIPTION
**Pull request by yshrsmz**
added DAOs

- CurrentEvaluationDao
- LatestEvaluationDao
- EventDao

implementation is based on v1 DAOs, but changed BLOB to TEXT as v2 uses JSON instead of Protocol Buffers.

If you want separate PR for each DAO, let me know.

### CODE REVIEW

File: mocks/build.gradle [hunk 1]
```diff
@@ -0,0 +1,20 @@
+plugins {
```
Comment (yshrsmz) [inline mocks/build.gradle:1]: `mocks` module is where all test mocks live.
It is `sharedTest` directory in v1, but it now shows a warning below and does not work correctly in Android Studio, so I created a dedicated module.

```
Duplicate content roots detected: Path [/Users/yshrsmz/repos/github.com/yshrsmz/bucketeer-android-sdk/bucketeer/src/sharedTest/java] of module [bucketeer-android-sdk.bucketeer.unitTest] was removed from modules [bucketeer-android-sdk.bucketeer.androidTest]
```

File: bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration1to2.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package io.bucketeer.sdk.android.internal.database.migration
+
+import androidx.sqlite.db.SupportSQLiteDatabase
+
+class Migration1to2 : Migration {
+
+  override fun migrate(db: SupportSQLiteDatabase) {
```
Comment (yshrsmz) [inline bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/database/migration/Migration1to2.kt:7]: simple drop & re-create migration.

**Review by cre8ivejp — APPROVED** [review]
Great!
Thank you!

**Comment by yshrsmz** [comment]
thanks

**Pull request merged**